crates/ruff_linter/src/rules/pyupgrade/rules/pep695/non_pep695_generic_class.rs

```diff
@@ -179,12 +187,41 @@ pub(crate) fn non_pep695_generic_class(checker: &mut Checker, class_def: &StmtCl
             source: checker.source(),
         };
 
-        diagnostic.set_fix(Fix::unsafe_edit(Edit::replacement(
-            type_params.to_string(),
-            name.end(),
-            arguments.end(),
-        )));
+        let Ok(edit) = remove_argument(
+            generic_expr,
+            arguments,
+            Parentheses::Remove,
+            checker.source(),
+        ) else {
+            return;
+        };
+
+        diagnostic.set_fix(Fix::unsafe_edits(
+            Edit::insertion(type_params.to_string(), name.end()),
+            [edit],
+        ));
```

Comment:
We have a `try_set_fix()` method that's handy for cases like this!

```suggestion
        diagnostic.try_set_fix(|| {
            let removal_edit = remove_argument(
                generic_expr,
                arguments,
                Parentheses::Remove,
                checker.source(),
            )?;
            Ok(Fix::unsafe_edits(
                Edit::insertion(type_params.to_string(), name.end()),
                [removal_edit],
            ))
        });
```

The advantage is that if you run Ruff with enough `-v` flags, it prints a useful error message to the terminal explaining why it was unable to provide a fix for the specific case where `remove_argument()` returned an error

Reply:
That is very handy! I was wondering how `remove_argument` could fail at all, so I'll be glad to see those messages if it does!